client/my-sites/plans-grid/types.ts

```diff
@@ -26,3 +34,72 @@ export type DataResponse< T > = {
 	isLoading: boolean;
 	result?: T;
 };
+
+export interface CommonGridProps {
+	/**
+	 * Site id may not be used in ComparisonGrid, but need to be investigated further
+	 */
+	siteId?: number | null;
+	isInSignup: boolean;
+	isLaunchPage?: boolean | null;
+	isReskinned?: boolean;
+	onStorageAddOnClick?: ( addOnSlug: WPComStorageAddOnSlug ) => void;
+	intervalType: string;
+	currentSitePlanSlug?: string | null;
+	hideUnavailableFeatures?: boolean; // used to hide features that are not available, instead of strike-through as explained in #76206
+	planActionOverrides?: PlanActionOverrides;
+
+	// Value of the `?feature=` query param, so we can highlight a given feature and hide plans without it.
+	selectedFeature?: string;
+	showUpgradeableStorage: boolean; // feature flag used to show the storage add-on dropdown
+	stickyRowOffset: number;
+	showRefundPeriod?: boolean;
+	// only used for comparison grid
+	planTypeSelectorProps?: PlanTypeSelectorProps;
+	onUpgradeClick: ( planSlug: PlanSlug ) => void;
+}
+
+export interface FeaturesGridProps extends CommonGridProps {
+	gridPlans: GridPlan[];
+	isLargeCurrency: boolean;
+	translate: LocalizeProps[ 'translate' ];
+	currentPlanManageHref?: string;
+	isPlanUpgradeCreditEligible: boolean;
+	generatedWPComSubdomain: DataResponse< { domain_name: string } >;
+	gridPlanForSpotlight?: GridPlan;
+	isCustomDomainAllowedOnFreePlan: DataResponse< boolean >; // indicate when a custom domain is allowed to be used with the Free plan.
+	paidDomainName?: string;
+	showLegacyStorageFeature: boolean;
+}
+
+export interface ComparisonGridProps extends CommonGridProps {
+	// Value of the `?plan=` query param, so we can highlight a given plan.
+	selectedPlan?: string;
+}
+
+export type GridContextProps = {
+	gridPlans: GridPlan[];
+	allFeaturesList: FeatureList;
+	intent?: PlansIntent;
+	usePricingMetaForGridPlans: UsePricingMetaForGridPlans;
+	children: React.ReactNode;
+};
+
+export type ComparisonGridExternalProps = Omit< GridContextProps, 'children' > &
```

Comment:
These are the ones the consumer would see when importing the respective component from the package right? Let's maybe add a `/** */` doc above each of these to clarify, I guess in case someone imports the type from here (I recall having to do that when rendering the component via async and needing the props type)